config/v1/types_feature.go

```diff
@@ -97,44 +97,98 @@ type FeatureGateEnabledDisabled struct {
-			"ServiceNodeExclusion",           // sig-scheduling, ccoleman
-		},
-		Disabled: []string{
-			"LegacyNodeRoleBehavior", // sig-scheduling, ccoleman
-		},
+	TechPreviewNoUpgrade: newDefaultFeatures().toFeatures(),
+	LatencySensitive: newDefaultFeatures().
+		with(
+			"TopologyManager", // sig-pod, sjenning
+		).
+		toFeatures(),
+}
+
+var defaultFeatures = &FeatureGateEnabledDisabled{
+	Enabled: []string{
+		"RotateKubeletServerCertificate", // sig-pod, sjenning
+		"SupportPodPidsLimit",            // sig-pod, sjenning
+		"NodeDisruptionExclusion",        // sig-scheduling, ccoleman
+		"ServiceNodeExclusion",           // sig-scheduling, ccoleman
+		"SCTPSupport",                    // sig-network, ccallend
 	},
-	LatencySensitive: {
-		Enabled: []string{
-			"RotateKubeletServerCertificate", // sig-pod, sjenning
-			"SupportPodPidsLimit",            // sig-pod, sjenning
-			"TopologyManager",                // sig-pod, sjenning
-			"NodeDisruptionExclusion",        // sig-scheduling, ccoleman
-			"ServiceNodeExclusion",           // sig-scheduling, ccoleman
-			"SCTPSupport",                    // sig-network, ccallend
-		},
-		Disabled: []string{
-			"LegacyNodeRoleBehavior", // sig-scheduling, ccoleman
-		},
+	Disabled: []string{
+		"LegacyNodeRoleBehavior", // sig-scheduling, ccoleman
 	},
 }
+
+type featureSetBuilder struct {
+	forceOn  []string
+	forceOff []string
+}
+
+func newDefaultFeatures() *featureSetBuilder {
+	return &featureSetBuilder{}
+}
+
+func (f *featureSetBuilder) with(forceOn ...string) *featureSetBuilder {
+	f.forceOn = append(f.forceOn, forceOn...)
+	return f
+}
+
+func (f *featureSetBuilder) without(forceOff ...string) *featureSetBuilder {
+	f.forceOff = append(f.forceOff, forceOff...)
+	return f
+}
+
+func (f *featureSetBuilder) isForcedOff(needle string) bool {
+	for _, forcedOff := range f.forceOff {
+		if needle == forcedOff {
+			return true
+		}
+	}
+	return false
+}
+
+func (f *featureSetBuilder) isForcedOn(needle string) bool {
+	for _, forceOn := range f.forceOn {
+		if needle == forceOn {
+			return true
+		}
+	}
+	return false
+}
+
+func (f *featureSetBuilder) toFeatures() *FeatureGateEnabledDisabled {
+	finalOn := []string{}
+	finalOff := []string{}
+
+	// only add the default enabled features if they haven't been explicitly set off
+	for _, defaultOn := range defaultFeatures.Enabled {
+		if !f.isForcedOff(defaultOn) {
+			finalOn = append(finalOn, defaultOn)
+		}
+	}
+	for _, currOn := range f.forceOn {
+		if f.isForcedOff(currOn) {
+			panic("coding error, you can't have features both on and off")
+		}
+		finalOn = append(finalOn, currOn)
+	}
+
+	// only add the default disabled features if they haven't been explicitly set on
+	for _, defaultOff := range defaultFeatures.Disabled {
+		if !f.isForcedOn(defaultOff) {
+			finalOff = append(finalOff, defaultOff)
+		}
+	}
+	for _, currOff := range f.forceOff {
+		finalOff = append(finalOff, currOff)
```

Comment:
do you think there is a need to check for coding errors here as well?

Reply:
> do you think there is a need to check for coding errors here as well?

we checked the other way, so you can't conflict, right?

Reply:
right, can't conflict